lib/seattleflu/id3c/cli/command/clinical.py

```diff
@@ -433,6 +424,203 @@ def add_kp_manifest_data(df: pd.DataFrame, manifest_filenames: tuple, manifest_f
+            map_sex(row['sex']),
+            str(row['birth_date']),
+            str(row["pat_address_zip"])
+        ), axis=1
+    )
+
+    clinical_records['identifier'] = clinical_records.apply(
+        lambda row: generate_hash(
+            f"{row['mrn']}{row['acc_num']}{row['collect_ts']}".lower()
+        ), axis=1
+    )
+
+    # calculate age based on sample collection date and birth day
+    clinical_records['birth_date'] = pd.to_datetime(clinical_records['birth_date'])
+    clinical_records['age'] = clinical_records.apply(
+        lambda row: age_ceiling(
+                relativedelta(
+                    row['collect_ts'],
+                    row['birth_date']
+                ).years
+            ), axis=1
+    )
+
+    # geocode addresses, then hash them and get the census tract to remove PII from downstream processes
+    with pickled_cache(geocoding_cache_file) as cache:
+        clinical_records['lat'], clinical_records['lang'], clinical_records['canonical_address'] = zip(
+            *clinical_records.apply(
+                lambda row: get_geocoded_address(
+                    {
+                        'street': row['pat_address_line1'],
+                        'secondary': row['pat_address_line2'],
+                        'city': row['pat_address_city'],
+                        'state': row['pat_address_state'],
+                        'zipcode': row['pat_address_zip']
+                    },
+                    cache
+                ),
+                axis=1
+            )
+        )
+
+    db = DatabaseSession()
+    clinical_records = clinical_records.apply(
+       lambda row: encode_addresses(db, row), axis=1
+    )
+
+    column_map = {
+        "collect_ts": "encountered",
+        "ethnic_group": "ethnicity",
+        'barcode': 'phskc_barcode'
+    }
+
+    columns_to_keep = list(column_map.keys()) + [
+        "_provenance",
+        "individual",
+        "identifier",
+        "site",
+        "sex",
+        "age",
+        "race",
+        "address",
+        "census_tract",
+        "main_cid",
+        "all_cids",
+        #"reason_for_visit",
+        "survey_testing_because_exposed",
+        "if_symptoms_how_long",
+        "survey_have_symptoms_now",
+        "inferred_symptomatic",
+        "vaccine_status",
+        "patient_class",
+        'encounter_status'
+    ]
+
+    clinical_records = clinical_records[columns_to_keep]
+    clinical_records = clinical_records.rename(columns=column_map)
+
+    # phskc data is sent with some rows duplicated, so before we add manifest data
+    # we should drop rows with duplicated values in their join criteria
+    clinical_records.drop_duplicates(subset=['phskc_barcode', 'main_cid', 'all_cids'], inplace=True)
+    clinical_records = add_phskc_manifest_data(clinical_records, phskc_specimen_manifest_filename)
+    clinical_records.drop('merge_col', axis=1, inplace=True)
+
+    dump_ndjson(clinical_records)
```

Comment:
We should also drop `main_cid`, `all_cids`, and `phskc_barcode` from the ndjson file, since they aren't needed after we have the sample barcode and we don't want to ingest them.